server/pbench/bin/index-pbench

```diff
@@ -801,14 +801,15 @@ class ToolData(object):
                     ('@metadata', self.run_metadata)
                 ])
                 datum[identifier][self.toolname] = _dict_const()
-                for klass in class_list.keys():
-                    datum[identifier][self.toolname][klass] = _dict_const([('id',identifier)])
                 try:
                     md = metadata[identifier]
                 except KeyError:
-                    pass
-                else:
-                    datum[identifier][self.toolname][klass].update(md)
+                    md = None
+                for klass in class_list.keys():
+                    d = _dict_const([('id',identifier)])
+                    if md:
+                        d.update(md)
+                    datum[identifier][self.toolname][klass] = d
```

Comment:
We need to document this stuff: at some point in the past I think I knew how this worked, but now I can't make heads or tails of it. I'll need to go back and start from scratch.

Reply:
Okay, i'll take a stab at documenting how this works.   Hopefully, documentation won't diverge from the code in the future!

Reply:
@ndokos, added code documentation to help make this more understandable.